### PR TITLE
enhancement: update feature's & feature list's `list_versions` method

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR updates `list_versions` method & documentation.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/api/feature.py [hunk 1]
```diff
@@ -374,18 +376,22 @@ def _list_versions(
...
         List saved Feature versions
 
-        >>> Feature.list_versions()  # doctest: +SKIP
-            name        version  dtype readiness  online_enabled             table    entities              created_at
-        0  new_feat2  V230323  FLOAT     DRAFT           False      [sf_event_table]  [customer] 2023-03-23 07:16:21.244
-        1  new_feat1  V230323  FLOAT     DRAFT           False      [sf_event_table]  [customer] 2023-03-23 07:16:21.166
-        2     sum_1d  V230323  FLOAT     DRAFT           False      [sf_event_table]  [customer] 2023-03-23 07:16:21.009
+        >>> Feature.list_versions(include_id=False)  # doctest: +ELLIPSIS
+                                     name  version      dtype         readiness  online_enabled                                          tables    primary_tables           entities   primary_entities  created_at  is_default
+        0  CustomerLatestInvoiceTimestamp      ...  TIMESTAMP  PRODUCTION_READY           False                                [GROCERYINVOICE]  [GROCERYINVOICE]  [grocerycustomer]  [grocerycustomer]         ...  True
```
Comment (YS-L) [inline featurebyte/api/feature.py:381]: Would it be better if we can check the full output? I think we can customise the clipping behaviour by setting the right pandas options maybe in conftest:

```
import pandas as pd
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)
```

File: .changelog/DEV-1802.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+change_type: enhancement
+
+# The name of the component
+component: list_versions
+
+# (Optional) One or more tracking issues or pull requests related to the change
+issues: []
+
+# A brief description of the change.
+note: "update Feature's & FeatureList's `list_versions` method"
```
Comment (YS-L) [inline .changelog/DEV-1802.yaml:11]: Would be good if the note can provide a bit more details on the changes.

File: featurebyte/api/feature.py (hunk 1 above)
Comment (gee-senbong) [reply featurebyte/api/feature.py:381]: This the full output. The main reason why `...` is used for the version is that it is sensitive to the date we run the test (version name contains date info). Including the version name will cause the test to pass only today but fail after that.

File: .changelog/DEV-1802.yaml (hunk 2 above)
Comment (gee-senbong) [reply .changelog/DEV-1802.yaml:11]: Will update this part to include info like adding `is_default` column. Any other things that you wish to include on the note?

**Review by YS-L — APPROVED** [review]

**Pull request merged**